bin/propolis-server/src/lib/migrate/source.rs

```diff
@@ -245,6 +261,16 @@ impl<T: AsyncRead + AsyncWrite + Unpin + Send> SourceProtocol<T> {
         }
     }
 
+    async fn time_data_read(&mut self) -> Result<(), MigrateError> {
+        let instance_guard = self.vm_controller.instance().lock();
+        let vmm_hdl = &instance_guard.machine().hdl;
+        let raw = vmm_hdl.export_vm()?;
```

Comment:
There's a subtlety in here that I think would be good to fix: `export_vm` returns a `MigrateStateError`, which has a `From<std::io:Error>` impl that converts all errors into `MigrateStateError::ImportFailed`, which is confusing since we're exporting here. (This hasn't been a problem before because entity state export is infallible.)

I'm not sure of the best way to fix this. We could remove the `From` impl but that would break all the uses of `?` in the import routines. Maybe we need a `StateExportError` enum for `export_vm` to use?

Reply:
Yeah, this is a great point. I took a stab at making error handling a bit better in cdbe3ef; there is a now a `MigrateError::TimeData` type that we convert all TimeData{Read} phase errors into. Let me know what you think.